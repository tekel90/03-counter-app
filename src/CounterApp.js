//para crear un functional component hay un snippet llamado
// rafcp

import React, { useState } from "react";
import PropTypes from "prop-types";

//el useState regresa un arreglo

//Functional Components
const CounterApp = ({ value = 0 }) => {
  const [counter, setCounter] = useState(value); //retorna un arreglo con 2 valores

  //handleAdd
  const handleAdd = () => {
    setCounter(counter + 1);
    // setCounter((c) => c + 1); //tomo el valor anterior del counter y le asigno c+1
  };

  //handleRemove
  const handleSubtract = () => {
    setCounter(counter - 1);
  };

  //handleReset
  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>

      <button onClick={handleAdd}> +1 </button>
      <button onClick={handleReset}> reset </button>
      <button onClick={handleSubtract}> -1 </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

export default CounterApp;

