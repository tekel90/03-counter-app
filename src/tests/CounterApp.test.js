import React from "react";
import "@testing-library/jest-dom";
import CounterApp from "../CounterApp";
import { shallow } from "enzyme";

describe("Pruebas en <CounterApp />", () => {
  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test("debe de mostar <CounterApp /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar el valor por defecto de 100", () => {
    const wrapper = shallow(<CounterApp value={100} />);
    const counterText = wrapper.find("h2").text().trim();

    expect(counterText).toBe("100");
  });

  test("debe incrementar el contador con el boton +1", () => {
    wrapper.find("button").at(0).simulate("click");
    const counterText = wrapper.find("h2").text().trim();
    expect(counterText).toBe("1");
  });

  test("debe decrementar el contador con el boton -1", () => {
    wrapper.find("button").at(2).simulate("click");
    const counterText = wrapper.find("h2").text().trim();
    expect(counterText).toBe("-1");
  });

  test("debe colocar el contador con el valor por defecto con el boton reset", () => {
    const wrapper = shallow(<CounterApp value={100} />);

    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(1).simulate("click");
    const counterText = wrapper.find("h2").text().trim();
    expect(counterText).toBe("100");
  });
});
