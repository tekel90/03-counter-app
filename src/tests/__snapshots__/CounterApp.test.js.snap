// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe de mostar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    onClick={[Function]}
  >
     reset 
  </button>
  <button
    onClick={[Function]}
  >
     -1 
  </button>
</Fragment>
`;
